### ISSUE TITLE
RubyLSP in windows 10 stuck on starting

### ISSUE THREAD

**Issue by motebaya** [Shopify/ruby-lsp#2369]
### Description

on Linux, it works fine without any issues, but for the past 2 days, i have been trying it on win10, and RubyLSP won't run on Vscode. it stuck at the 'starting' state. 

i thought it was a duplicate of #2286, but it's not; this is just a simple project with a `Gemfile` file: https://github.com/Shopify/ruby-lsp/issues/2286#issuecomment-2223529470

ruby version: [ruby 3.3.4 (2024-07-09 revision be1089c8ec) [x64-mingw-ucrt]](https://github.com/oneclick/rubyinstaller2/releases/download/RubyInstaller-3.3.4-1/rubyinstaller-devkit-3.3.4-1-x64.exe)
ruby-lsp: 
```
➜ bundle info ruby-lsp
  * ruby-lsp (0.17.9)
        Summary: An opinionated language server for Ruby
        Homepage: https://github.com/Shopify/ruby-lsp
        Documentation: https://shopify.github.io/ruby-lsp/                                             
        Path: D:/Software/Ruby33-x64/lib/ruby/gems/3.3.0/gems/ruby-lsp-0.17.9
```

# what i do:

<details>
    <summary>create folder/file and open it with powershell</summary>

    mkdir tryruby
    cd tryruby
    touch main.rb
    bundle init
    bundle add httparty --group production
    bundle add ruby-lsp --group development
    code .
    
</details>

<details>
    <summary>Gemfile </summary>
    ```
    # frozen_string_literal: true
    
    source "https://rubygems.org"
    
    # gem "rails"
    
    gem "httparty", "~> 0.22.0", :group => :production
    
    gem "ruby-lsp", "~> 0.17.9", :group => :development
    ```
</details>

<details>
<summary>Gemfile.lock</summary>

      GEM
        remote: https://rubygems.org/
        specs:
          bigdecimal (3.1.8)
          csv (3.3.0)
          httparty (0.22.0)
            csv
            mini_mime (>= 1.0.0)
            multi_xml (>= 0.5.2)
          language_server-protocol (3.17.0.3)
          logger (1.6.0)
          mini_mime (1.1.5)
          multi_xml (0.7.1)
            bigdecimal (~> 3.1)
          prism (0.30.0)
          rbs (3.5.2)
            logger
          ruby-lsp (0.17.9)
            language_server-protocol (~> 3.17.0)
            prism (>= 0.29.0, < 0.31)
            rbs (>= 3, < 4)
            sorbet-runtime (>= 0.5.10782)
          sorbet-runtime (0.5.11495)
      
      PLATFORMS
        x64-mingw-ucrt
      
      DEPENDENCIES
        httparty (~> 0.22.0)
        ruby-lsp (~> 0.17.9)
      
      BUNDLED WITH
         2.5.16

</details>

<details>
<summary>main.rb</summary>

    require 'httparty'
    ip = HTTParty.get('https://httpbin.org/ip', {
      headers: {
        'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36'
      }
    }).body
    puts ip

</details>

## output tab in VScode
![Screenshot 2024-07-26 170700](https://github.com/user-attachments/assets/85ca08d1-06d1-4aba-a660-45ab95f5c35e)

## in powershell
![image](https://github.com/user-attachments/assets/fa3542e9-0951-48db-8ee7-b206ce2ca545)

i don't know what's going on here, even after completely uninstalling VSCode and reinstalling it, nothing changed and it's still the same.

**Comment by motebaya** [comment]
> Thank you for the bug report. It seems that you're using the `custom` version manager, but in the output tab no command was printed. If you look at the first line of the first screenshot, the command we tried to run was ` && ruby ...`, which makes me think there's no command given the empty space before the `&&`.
> 
> How are you setting up the version manager? If you completely remove all version manager configuration and either rely on the RubyInstaller integration or select the Ruby version manually, does it work?

I have tried using rbenv from https://github.com/ccmywish/rbenv-for-windows because i thought it was the only one I could find for Windows, but a different issue came;

setup rbenv in $PROFILE following docs and reinstall ruby
![image_2024-07-27_00-50-47](https://github.com/user-attachments/assets/575fcc57-c269-4dd1-ab43-f398b19f1b20)
![Screenshot 2024-07-27 013101](https://github.com/user-attachments/assets/ab6c31fc-5d89-45c1-bb02-51e5a24a0191)


i also adding RUBY_VERSION to env path .
but, here the log from output tab vscode:
![Screenshot 2024-07-27 011304](https://github.com/user-attachments/assets/e311a73f-5857-4d7c-bcb4-e884e6d96f89)

I don't know why the ruby version is still not detected and shows as undefined. i have even created a `.ruby_version` file with `3.3.3-1` in current project folder and added `RUBY_VERSION` to the ENV path.

full log: 
[log.txt](https://github.com/user-attachments/files/16395420/log.txt)

The linter and snippets work as well, but autocomplete doesn't work at all (because LSP is not running yet). 
![Screenshot 2024-07-27 012644](https://github.com/user-attachments/assets/4f108308-7da5-4356-a3b3-ad9be17dd42a)


If there's anything needed to investigate the issue, please let me know. I'll provide it.!

**Comment by motebaya** [comment]
Closed this, because the issue has been resolved. i think the problem was the default terminal in VSCode not being compatible. i tried switching it to Git Bash, but that didn't help. Then, when I switched it to Windows PowerShell (`C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe`), it worked perfectly! I don't even know why!
![Screenshot 2024-07-27 142947](https://github.com/user-attachments/assets/1d413863-31e6-45b8-b50a-0c276a3b052c)

![Screenshot 2024-07-27 144049](https://github.com/user-attachments/assets/e9ed08af-c0a0-4d35-98e1-5a13efa5ed07)
![Screenshot 2024-07-27 144019](https://github.com/user-attachments/assets/6004b8b9-ba4f-4fc9-9821-de06c55d97cd)
